capa/rules/cache.py

```diff
@@ -26,6 +28,65 @@
 CacheIdentifier = str
 
 
+def is_dev_environment() -> Optional[Path]:
+    if getattr(sys, "frozen", False):
+        # running as a PyInstaller executable
+        return None
+
+    if "site-packages" in __file__:
+        # running from a site-packages installation
+        return None
+
+    capa_root = Path(__file__).resolve().parent.parent
+    git_dir = capa_root / ".git"
+
+    if not git_dir.is_dir():
+        # .git directory doesn't exist
+        return None
+
+    git_exe = shutil.which("git")
+    if not git_exe:
+        # git is not found in PATH
+        return None
+
+    return Path(git_exe)
+
+
+def get_modified_files() -> List[Path]:
+    try:
+        # use git status to retrieve tracked modified files
+        result = subprocess.run(
+            ["git", "--no-pager", "status", "--porcelain", "--untracked-files=no"],
```

Comment:
does this also work across OS?

Reply:
[`Subprocess`](https://docs.python.org/3/library/subprocess.html) is compatible with Windows and POSIX-compliant platforms (Linux/Mac OS X)

Reply:
I mean `git` (should it be `git.exe` on Windows?)

Reply:
`.exe` file extention is part of the default `%PATHEXT%` on Windows, this means windows will run `git.exe` without having to specifying the file extention. See [here](https://superuser.com/a/1330371).

Reply:
I think we can even avoid this, and set `git_exe` to the return value of `shutil.which('git')`. This guarantees that we get the full path of `git` which ends with `.exe`.